client/client.go

```diff
@@ -35,10 +35,14 @@ import (
 	"github.com/snapcore/snapd/dirs"
 )
 
-func unixDialer(_, _ string) (net.Conn, error) {
+func privateUnixDialer(_, _ string) (net.Conn, error) {
```

Comment:
Neither of these sockets are private to any particular user. They are world readable and writable. Snaps cannot access one of them, as you point out. That's indeed the diverging point, and where my suggestion comes from.

Would you prefer to name the socket as snapd-snap.socket?  That sounds reasonable as well. Or some other suggestion?